src/Impostor.Api/Net/Messages/C2S/HandshakeC2S.cs

```diff
@@ -1,12 +1,28 @@
+using Impostor.Api.Innersloth;
+
 namespace Impostor.Api.Net.Messages.C2S
 {
     public static class HandshakeC2S
     {
-        public static void Deserialize(IMessageReader reader, out int clientVersion, out string name, out uint lastNonceReceived)
+        public static void Deserialize(IMessageReader reader, out int clientVersion, out string name, out uint lastNonceReceived, out Language language, out QuickChatModes chatMode)
         {
             clientVersion = reader.ReadInt32();
             name = reader.ReadString();
             lastNonceReceived = reader.ReadUInt32();
+
+            // Version 2021.6.30 (aka 50537300) is the first version that includes language and chat modes
+            // In case the game is older than that, stop reading here.
+            if (clientVersion >= 50537300)
```

Comment:
Use `GameVersion.GetVersion(2021, 6, 30)` for cleaner code? It's just a few simple math operations so performance shouldn't matter that much, eventually cache in a private readonly static field